### DISCUSSION TITLE
Rate Drop to Min

### DISCUSSION THREAD

**Discussion by Fail-Safe** [sqm-autorate/sqm-autorate#175] [Algorithm Design]
@dlakelan Not sure if you're still following anything here. But if so, I keep seeing this behavior where my rate drops back to min for no apparent reason (at least according to my attached graphs). I'm trying to remember the cases where this occurs. I'll keep digging, but if you feel inclined to look and comment, I'd love to know your thoughts on it :)

<img width="515" alt="image" src="https://github.com/sqm-autorate/sqm-autorate/assets/10307870/1ba6eafe-dab9-4a46-a9b1-ab3b855b9823">

**Comment by chas-iot** [comment]
@Fail-Safe 

To my memory, this is functioning as originally designed. When there is a spike in the delay while the throughput load is low, then the rate is set to the minimum. Which should quickly recover when the throughput increases, assuming that the increased delay was transient.

With the delay-histogram plugin running, then this drastic cut is mitigated to halfway between the current rate and the minimum rate.

Regards,
Charles

**Reply by Fail-Safe** [reply]
That's ringing a bell now--thank you! I have been running your delay-histogram plugin and it's really slick!

The sudden rate cut is actually proving to be more disruptive than I had originally noticed. It is enough to cause a momentary hit to any realtime communications I have going on (MS Teams, WebEx, etc).
